### ISSUE TITLE
Computed property observer not fired while @shared property change

### ISSUE THREAD

**Issue by marcalexiei** [ractivejs/ractive#3184]
## Description:
Sometimes `computed` property observer handler is not called if the computed property observes data inside `@shared` keypath.

Maybe issue appears because `@shared.query` is set both time to empty object.

## Workaround
Property is accessible inside component or in parents component data "scope".

## Versions affected:
Ractive `0.9.x` and `edge`.

## Platforms affected:
All

## Reproduction:
https://jsfiddle.net/gu34hfm3/

1. Observe "OUTPUT" messages.
2. Last computed property change doesn't fire the observer callback.

```javascript
Ractive.components.Page = Ractive.extend({
  template: `
  <section>
		{{activeTab.label}}
  </section>
  `,

  data() {
    return {
      tabSet: [
        {
          name: 'tab1',
          label: 'Tab 1',
          isActive: ({ tab }) => (!tab || tab === 'tab1'),
        },
        {
          name: 'tab2',
          label: 'Tab 2',
          isActive: ({ tab }) => (tab === 'tab2'),
        },
      ],
    };
  },

  computed: {
    activeTab() {
      const tabSet = this.get('tabSet');
      const route = this.get('@shared.route');
      const query = this.get('@shared.query');

      const activeTab = tabSet.find(t => t.isActive(route, query));
      this.root.push('messages', `computed property calculation: ${JSON.stringify(activeTab)}`);
      return activeTab;
    },
  },

  observe: {
    activeTab(activeTab) {
      if (activeTab) {
        this.root.push('messages', `observer fired: ${JSON.stringify(activeTab)}`);
        this.root.push('messages', '----------------------------');
      }
    },
  },
});


const r = window.r = new Ractive({
  template: `
  <h1>Computed property not fired while @shared property change</h1>
  {{#if showComponent}}
  	<Page/>
  {{/if}}
  
  <h2>OUTPUT</h2>
  {{#each messages}}
  	<p>{{.}}</p>
  {{/each}}
  `,
  data: {
    showComponent: false,

    messages: [],
  }
});

r.set({ '@shared.route': { tab: 'tab1' }, '@shared.query': {} });

r.render('#main').then(() => {
  r.set({ showComponent: true }).then(() => {
    setTimeout(() => {
      r.set({ '@shared.route': { tab: 'tab2' }, '@shared.query': {} });
    }, 500);
  });
});
```

**Comment by PaulMaly** [comment]
Hm, I'm not sure that is ever was. But seems it's could be useful in many cases.

**Comment by evs-chris** [comment]
The multi-model update in the setTimeout turns out to be the cause of the issue. The computation gets updated twice, but the observer was only allowing one update per runloop turn. This is now fixed in edge and will go out with 0.9.12. Thanks for the report!
